capi/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+cmake_minimum_required (VERSION 3.0.2...3.28.1)
+
+SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
+SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+
+project(libqibolab)
+
+set(VERSION "\"0.0.1\"")
+
+find_package(Python3 COMPONENTS Interpreter Development)
```

Comment:
For some reason, it is able to find the interpreter, and it doesn't complain
```
-- Found Python3: /Users/alessandro/Projects/Quantum/qibolab/.venv/bin/python3.11 (found version "3.11.7") found components: Interpreter Development Development.Module Development.Embed
generating cqibolab.c
(already up-to-date)
-- Configuring done (3.2s)
-- Generating done (0.0s)
```
but then I'm missing the Python headers, and it crashes during the actual build
<details><summary>Build error</summary>
<p>

```
❯ cmake --build build
[ 50%] Building C object CMakeFiles/qibolab.dir/src/cqibolab.c.o
[100%] Linking C shared library libqibolab.dylib
Undefined symbols for architecture arm64:
  "_PyCapsule_Type", referenced from:
      __cffi_carefully_make_gil in cqibolab.c.o
  "_PyDict_GetItemString", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyDict_New", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyDict_SetItemString", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyErr_Display", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyErr_Fetch", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyErr_NormalizeException", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyErr_Occurred", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyEval_EvalCode", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyEval_GetBuiltins", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyEval_SaveThread", referenced from:
      __cffi_carefully_make_gil in cqibolab.c.o
  "_PyFile_WriteObject", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyFile_WriteString", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyGILState_Ensure", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyGILState_Release", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyImport_GetModuleDict", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyImport_ImportModule", referenced from:
      __cffi_init in cqibolab.c.o
  "_PyLong_FromVoidPtr", referenced from:
      __cffi_init in cqibolab.c.o
  "_PyObject_CallMethod", referenced from:
      __cffi_init in cqibolab.c.o
  "_PyObject_Free", referenced from:
      __cffi_free_array_arguments in cqibolab.c.o
  "_PyObject_GetAttrString", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_PyObject_Malloc", referenced from:
      __cffi_convert_array_argument in cqibolab.c.o
  "_PySys_GetObject", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_Py_CompileStringExFlags", referenced from:
      __cffi_initialize_python in cqibolab.c.o
  "_Py_InitializeEx", referenced from:
      __cffi_py_initialize in cqibolab.c.o
  "_Py_IsInitialized", referenced from:
      __cffi_carefully_make_gil in cqibolab.c.o
  "__Py_Dealloc", referenced from:
      _Py_DECREF in cqibolab.c.o
  "__Py_NoneStruct", referenced from:
      __cffi_initialize_python in cqibolab.c.o
ld: symbol(s) not found for architecture arm64
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/qibolab.dir/build.make:97: libqibolab.dylib] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/qibolab.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

</p>
</details>

(of course I'm not using Ubuntu, but Nix... for which I'm trying to fix by myself...)

Reply:
Should be fixed now.

Reply:
I was planning to make the workflow myself, but I first I wanted to compile it (on MacOS, with Nix... it's not being the easiest...)

Reply:
Ok, sounds reasonable.